core/services/log/registrations.go

```diff
@@ -0,0 +1,283 @@
+package log
+
+import (
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+)
+
+// How it works in general:
+//
+// 1. Adding and removing listeners updates the highestNumConfirmations - a number tracking what's the current highest NumConfirmations globally
```

Comment:
What is NumConfirmations? Is it the same as block confirmations? Why is it affected by adding/removing listeners?

Reply:
Can each listener specify its own minimum required number of confirmations?

Reply:
What does this minimum number of confirmations do?

Reply:
`Can each listener specify its own minimum required number of confirmations?` - yes, pretty much - that's NumConfirmations - the main purpose of this PR.

This global highestNumConfirmations is just used internally for keeping track when it's safe to delete logs from the array

Reply:
`What is NumConfirmations? Is it the same as block confirmations? ` - yes, number of heads that need to be received before a log from an older block can be sent